src/Illuminate/Database/Eloquent/Concerns/HasTimestamps.php

```diff
@@ -155,4 +155,21 @@ public function getQualifiedUpdatedAtColumn()
     {
         return $this->qualifyColumn($this->getUpdatedAtColumn());
     }
+
+    /**
+     * Run the given callable without timestamping the model.
+     *
+     * @param  callable  $callback
+     * @return mixed
+     */
+    public function withoutTimestamps(callable $callback)
+    {
+        if (! $this->usesTimestamps()) {
+            return $callback($this);
+        }
+
+        $this->timestamps = false;
+
+        return tap($callback($this), fn () => $this->timestamps = true);
```

Comment:
Copying `withoutEvents()` behaviour, I think this should handle callback throwables like `QueryException`. Otherwise `$timestamps = true` may not be restored.

```suggestion
        try {
            return $callback($this);
        } finally {
            $this->timestamps = true;
        }
```

Reply:
Updated and tested. Thanks!